updater/lib/dependabot/updater.rb

```diff
@@ -81,18 +80,20 @@ def run
           "Dependabot::AllVersionsIgnored was unexpectedly raised for a non-security update job"
         )
         error.set_backtrace(e.backtrace)
-        Raven.capture_exception(error, raven_context)
+        service.capture_exception(error: error, job: job)
         return
       end
 
       # OOM errors are special cased so that we stop the update run early
-      error = { "error-type": RUN_HALTING_ERRORS.fetch(e.class) }
-      record_error(error)
+      service.record_update_job_error(
+        error_type: RUN_HALTING_ERRORS.fetch(e.class),
```

Comment:
🔎 I see we weren't handling this before, so no need to change anything. Should we provide a fallback value for `Hash#fetch` here?

Reply:
I think we probably want it to fail loud as we, ideally, should never enter this method unless we already checked the subject `error` was present in `RUN_HALTING_ERRORS` to begin with.

This code might be able to fold up into the call site in another pass?